File: Algorithm/Hash/programmers_완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    unordered_map<string, int> umap;
    
    for (const string par: participant)
        umap[par]++;
    
    for (const string com: completion)
        umap[com]--;
    
    for (auto pair : umap) {
        if (pair.second)
            answer = pair.first;
    }
    
    return answer;
}

File: Algorithm/Hash/programmers_완주하지 못한 선수.py
from collections import Counter

def solution(participant, completion):
    answer = ''
    players = Counter(participant)
    
    for comp in completion:
        players[comp] -= 1
    
    for (k,v) in players.items():
        if v > 0:
            answer = k
            break
    
    return answer